go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -0,0 +1,389 @@
+		for _, addressWanted := range addressesWanted {
+			var exists bool
+			for _, addressFound := range addressesFound {
+				if addressFound.Equal(addressWanted) {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				// Use arping to try to update other hosts ARP caches, in case this IP was
+				// previously active on another node. (Based on code from "ifup".)
+				go func() {
+					out, err := exec.Command("/sbin/arping", "-q", "-A", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).CombinedOutput()
+					if err != nil {
+						klog.Warningf("Failed to send ARP claim for IP %s: %v (exec command output: %q)", addressWanted.IP.String(), err, string(out))
+						return
+					}
+					time.Sleep(2 * time.Second)
+					_ = exec.Command("/sbin/arping", "-q", "-U", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).Run()
+				}()
+
+				if err = netlinkOps.AddrAdd(link, &addressWanted); err != nil {
+					klog.Errorf("Link manager: failed to add address %q to link %q: %v",
+						addressWanted.String(), link.Attrs().Name, err)
+				}
+				// Use arping to try to update other hosts ARP caches, in case this IP was
+				// previously active on another node. (Based on code from "ifup".)
+				go func() {
+					out, err := exec.Command("/sbin/arping", "-q", "-A", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).CombinedOutput()
+					if err != nil {
+						klog.Warningf("Failed to send ARP claim for IP %s: %v (exec command output: %q)", addressWanted.IP.String(), err, string(out))
+						return
+					}
+					time.Sleep(2 * time.Second)
+					_ = exec.Command("/sbin/arping", "-q", "-U", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).Run()
+				}()
+				klog.Infof("Link manager completed adding address %s to link %s", addressWanted, link.Attrs().Name)
+			}
+		}
+	}
+}
+
+func (c *Controller) addAddressToStore(linkName string, newAddress netlink.Addr) {
+	addressesSaved, found := c.store[linkName]
+	if !found {
+		c.store[linkName] = []netlink.Addr{newAddress}
+		return
+	}
+	// check if the address already exists
+	for _, addressSaved := range addressesSaved {
+		if addressSaved.Equal(newAddress) {
+			return
+		}
+	}
+	// add it to store if not found
+	c.store[linkName] = append(addressesSaved, newAddress)
+}
+
+func (c *Controller) delAddressFromStore(linkName string, address netlink.Addr) {
+	addressesSaved, found := c.store[linkName]
+	if !found {
+		return
+	}
+	temp := addressesSaved[:0]
+	for _, addressSaved := range addressesSaved {
+		if !addressSaved.Equal(address) {
+			temp = append(temp, address)
+		}
+	}
+	c.store[linkName] = temp
+}
+
+// GetAssignedAddressLabel returns the label that must be assigned to each egress IP address bound to an interface
+func GetAssignedAddressLabel(linkName string) string {
+	//TODO: undeterstand why label char limt is 15 and how can a label work with a long link name otherwise it may block eip assignment
```

Comment:
nit: understand

Reply:
should we try to hash the link name to 12 characters, and suffix with ovn after that? Does the label need to have the link name?

Reply:
no the label doesnt have to be the link name - link name hashed and limited to 15.

Reply:
Ah I was right originally :) The label must start with the interface name:
```
go-controller/vendor/github.com/vishvananda/netlink/addr_linux.go:79:		return fmt.Errorf("label must begin with interface name")
```
I will test manually a very long label name and ensure its not blocking.